spec/Section 6 -- Execution.md

```diff
@@ -649,10 +835,26 @@ CompleteValue(fieldType, fields, result, variableValues):
   * If {result} is {null} (or another internal value similar to {null} such as
     {undefined} or {NaN}), return {null}.
   * If {fieldType} is a List type:
+    * If {result} is an iterator:
```

Comment:
We have now a rough implementation of stream for hotchocolate and I ran into some things that are not so clear. 

When streaming a field that has a list of composite types the iterator will yield objects like the following:

```json
{
  "id": "1000",
   "name": "Luke Skywalker"
}
```

These will than be integrated into a result that could look like the following:

```json
{
  "path": ["hero", "friends", "nodes", 1],
  "data": {
    "id": "1000",
    "name": "Luke Skywalker"
  },
  "hasNext": false
}
```

This works quite well... But what if the list is a list of leaf values like a list of strings? I have the same issue with a nested list where the element is actually a list of something.

Can data in this case be a scalar or a list? Would this break the contract for GraphQL results?

Reply:
A list of strings would just be:

```json
{
  "path": ["hero", "friends", "items"],
  "data": ["hi there"],
  "hasNext": false
}
```

i suspect youre talking about `@stream` there — then:


```json
{
  "path": ["hero", "friends", "nodes", 0],
  "data": "hi there",
  "hasNext": false
}
```

Reply:
sure... but so far `data` was guaranteed by the spec to be `null` or a map. We need to clarify this.

Reply:
Maybe we should also put some examples in for people who are implementing this.

Reply:
I opened a discussion for this here: https://github.com/robrichard/defer-stream-wg/discussions/21